File: node-backend/src/models/Event.js
import mongoose from 'mongoose';

// Define the Event schema
const eventSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: [true, 'Event name is required'],
    },
    description: {
      type: String,
      default: '',
    },
    dateTime: {
      type: Date,
      required: [true, 'Event date and time are required'],
    },
    location: {
      type: String,
      required: [true, 'Event location is required'],
    },
    participants: {
      type: [String],
      default: [],
    },
    images: {
      type: [String],
      default: [],
    },
    postedBy: {
      type: String,
      default: '',
    },
  },
  {
    timestamps: true,
  }
);

// Create and export the Event model
const Event = mongoose.model('Event', eventSchema);
export default Event;

File: java-backend/src/main/java/fr/epita/eventure/models/Event.java
package fr.epita.eventure.models;

import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;
import java.time.LocalDateTime;

@Setter
@Getter
@Document(collection = "events")
public class Event {

    @Id
    private String id;

    @NotBlank(message = "Event name is required")
    private String title;

    private String description;

    @NotBlank(message = "Event name is required")
    private LocalDateTime dateTime;

    @NotBlank(message = "Event name is required")
    private String location;

    private List<String> participants;

    private List<String> images;

    private String postedBy;
}
